src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,405 @@
+        Map<String, Double> participationPerBus;
+        if (lfParameters.isDistributedSlack()) {
+            participatingElements = getParticipatingElements(lfNetwork, lfParameters);
+            participationPerBus = participatingElements.stream().collect(Collectors.toMap(
+                element -> getParticipatingElementBus(element).getId(),
+                element -> -element.getFactor(),
+                Double::sum
+            ));
+        } else {
+            participationPerBus = Collections.singletonMap(lfNetwork.getSlackBus().getId(), -1d);
+        }
+        // compute the participation for each injection factor (+1 on the injection, and then -participation on all
+        // elements participating to load balancing)
+        computeFactorsInjection(participationPerBus, factorsGroups);
+
+        Map<String, ComputedContingencyElement> contingenciesElements = contingencies.stream()
+                                                                                    .flatMap(contingency -> contingency.getElements().stream())
+                                                                                    .map(contingencyElement -> new ComputedContingencyElement(contingencyElement, lfNetwork, equationSystem))
+                                                                                    .collect(Collectors.toMap(
+                                                                                        computedContingencyElement -> computedContingencyElement.getElement().getId(),
+                                                                                        computedContingencyElement -> computedContingencyElement
+                                                                                    ));
+
+        ComputedContingencyElement.setGlobalIndexes(contingenciesElements.values(), factorsGroups.size());
 
         // create jacobian matrix either using base network calculated voltages or nominal voltages
         VoltageInitializer voltageInitializer = lfParameters.getVoltageInitMode() == LoadFlowParameters.VoltageInitMode.PREVIOUS_VALUES ? new PreviousValueVoltageInitializer()
                 : new UniformValueVoltageInitializer();
         JacobianMatrix j = createJacobianMatrix(equationSystem, voltageInitializer);
+        // compute pre-contingency sensitivity values + the states for making +1-1 on the contingencies
+        DenseMatrix rhs = initRhs(lfNetwork, equationSystem, factorsGroups, contingenciesElements);
+        DenseMatrix states = solveTransposed(rhs, j); // states contains angles for the sensitivity factors, but also for the +1-1 related to contingencies
+
+        // sensitivities without contingency
+        List<SensitivityValue> sensitivityValues = calculateSensitivityValues(factorsGroups,
+                states, Collections.emptyList());
+
+        Collection<Contingency> straightforwardContingencies = new LinkedList<>();
```

Comment:
`straightforwardContingencies` should be renamed to reflect that these contingencies do not loose connectivity.